File: app/src/main/java/linusfessler/alarmtiles/timer/TimerDao.java
package linusfessler.alarmtiles.timer;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface TimerDao {

    @Query("SELECT * FROM Timer")
    LiveData<Timer> select();

    @Insert
    void insert(Timer timer);

    @Update
    void update(Timer timer);

    @Query("SELECT COUNT(*) FROM Timer")
    int count();
}

File: app/src/main/java/linusfessler/alarmtiles/timer/TimerDao.kt
package linusfessler.alarmtiles.timer

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update

@Dao
interface TimerDao {
    @Query("SELECT * FROM Timer")
    fun select(): LiveData<Timer>

    @Insert
    fun insert(timer: Timer)

    @Update
    fun update(timer: Timer)

    @Query("SELECT COUNT(*) FROM Timer")
    fun count(): Int
}